### ISSUE TITLE
Wrong Archive String on Zorin OS

### ISSUE THREAD

**Issue by damms005** [nodkz/mongodb-memory-server#616]
## Versions

- NodeJS: 17+
- mongodb-memory-server: 8+
- system: Linux (Zorin 16.0)

package: mongo-memory-server

## What is the Problem?
On download of mongo db archive from [https://fastdl.mongodb.org](https://fastdl.mongodb.org), with debug mode turned on, the os in console log looks like:
```
os: {
    os: 'linux',
    dist: 'zorin',
    codename: 'focal',
    release: '16',
    id_like: [ 'ubuntu' ]
  }
```

However, the generated archive string is like `linux/mongodb-linux-x86_64-ubuntu1604-4.7.0.tgz`. The error here is `ubuntu1604` in that archive string suggests that the download is for *Ubuntu 16.04*. However, Zorin 16.0 is based on *Ubuntu 20.04*

## Do you know *why* it happens?

*no*. However, the installer somehow wrongly uses an old version of Ubuntu in the archive string, and hence downloads old version of Mongo DB on Zorin OS

**Comment by hasezoey** [comment]
please provide the output for the following 3 commands:

```sh
$ cat /etc/lsb-release
$ cat /etc/os-release
$ cat /etc/upstream-release/lsb-release
```

**Comment by damms005** [comment]
```
❯ cat /etc/lsb-release
DISTRIB_ID=Zorin
DISTRIB_RELEASE=16
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Zorin OS 16.1"
```

```
❯ cat /etc/os-release
NAME="Zorin OS"
VERSION="16.1"
ID=zorin
ID_LIKE=ubuntu
PRETTY_NAME="Zorin OS 16.1"
VERSION_ID="16"
HOME_URL="https://zorin.com/os/"
SUPPORT_URL="https://help.zorin.com/"
BUG_REPORT_URL="https://zorin.com/os/feedback/"
PRIVACY_POLICY_URL="https://zorin.com/legal/privacy/"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```


```
❯ cat /etc/upstream-release/lsb-release
cat: /etc/upstream-release/lsb-release: No such file or directory
```

**Comment by hasezoey** [comment]
thanks for this information, currently if we dont have any special handling for a distro, but it has a `ID_LIKE`, it will be passed to the handler for `ID_LIKE`'s distro, which currently only transforms the input if it is also specially handled (see linux mint), so it is currently basically treating Zorin like ubuntu without transformation and inital input (which is a "bug")

tough unless Zorin actually has these codenames like ubuntu, it is weird that they set it as a codename
also, if you have access to Zorin's developers, maybe open a issue there to have a `/etc/upstream-release/lsb-release` set-up and remove `VERSION_CODENAME` / `DISTRIB_CODENAME` if this is not the actual codename for the version

**Comment by hasezoey** [comment]
A fix is in place, which will ship with 8.4.1 (which will be automatically published after the build, in around ~20 minutes from this post)

though this will only prevent wrong values, but will not have special mapping for Zorin OS, for this please open a new issue as `Distribution Support` or a direct PR implementing a mapping (or better ask the Zorin OS maintainers to have a upstream file)
